qiskit_aer/quantum_info/states/aer_statevector.py

```diff
@@ -126,10 +126,13 @@ def sample_memory(self, shots, qargs=None):
         self._aer_state.close()
 
         self._aer_state.renew()
-        self._aer_state.initialize(self._data, copy=False)
+        copied = self._aer_state.initialize(self._data, copy=False)
 
         samples = self._aer_state.sample_memory(qubits, shots)
-        self._data = self._aer_state.move_to_ndarray()
+        if copied:
+            self._aer_state.move_to_ndarray()
+        else:
+            self._data = self._aer_state.move_to_ndarray()
```

Comment:
I think this `if` might be backwards?  All the paths for `AerState.initialize` that existed before this PR now return `True`, which now means that `self._data` is no longer assigned to, so it's a behavioural change.  That said, I don't understand why we're only conditionally assigning; I think it's safe just to always assign?

Reply:
sorry. this branch should not be necessary...